bioblend/galaxy/libraries/__init__.py

```diff
@@ -96,6 +96,37 @@ def delete_library_dataset(self, library_id, dataset_id, purged=False):
         url = '/'.join([url, dataset_id])
         return self._delete(payload={'purged': purged}, url=url)
 
+    def update_library_dataset(self, library_id, dataset_id, **kwds):
+        """
+        Update library dataset metadata. Some of the attributes that can be
+        modified are documented below.
+
+        :type library_id: str
+        :param library_id: library id where dataset is found in
+
+        :type dataset_id: str
+        :param dataset_id: id of the dataset to be deleted
+
+        :type name: str
+        :param name: Replace library dataset name with the given string
+
+        :type misc_info: str
+        :param misc_info: Replace library dataset misc_info with given string
+
+        :type file_ext: str
+        :param file_ext: Replace library dataset extension (must exist in the Galaxy registry)
+
+        :type genome_build: str
+        :param genome_build: Replace library dataset genome build (dbkey)
+
+        :rtype: dict
+        :return: details of the updated dataset
+        """
+        url = self.gi._make_url(self, library_id, contents=True)
+        # Append the dataset_id to the base history contents URL
+        url = '/'.join([url, dataset_id])
```

Comment:
This will set `url` to `<base_url>/api/libraries/<library_id>/contents/<dataset_id>`, which executes this: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/library_contents.py#L373 , which is not the usual Galaxy `update` API, but has been there [since 2011](https://github.com/galaxyproject/galaxy/commit/38d26e946e6b0e2438ffc25e09923938860b5b35).

The `url` you want is `<base_url>/api/libraries/datasets/<dataset_id>` , i.e.:
```python
url = '/'.join([self.gi._make_url(self), 'datasets', dataset_id])
```

You can also remove `library_id` from the method parameters.